File: 2_semester/Laba_4/task_2.cpp
#include <iostream>
#include <vector>
#include <random>
#include <ctime>

std::vector<int> generator(int count){
	std::minstd_rand gen;
	gen.seed(time(0));
	std::vector<int> main_list;
	for(int i = 0; i < count; i++){
		main_list.push_back(gen()%200-100);
	}
	return main_list;
}

void sort(std::vector<int> &main_list, int first, int second){
	int temp;
	for (int i = first; i <= second; i++)
		for (int j = i; j <= second; j++)
			if (main_list[i] < main_list[j]){
				temp = main_list[i];
				main_list[i] = main_list[j];
				main_list[j] = temp;
			}
}

int main(){
	int count = 10, a = 2, b = 8;
	std::vector<int> main_list, sorted_list;
	//std::cin >> count;	
	//std::cin >> a;	
	//std::cin >> b;	

	main_list = generator(count);
	for (auto element : main_list){
		std::cout << element << " ";
	}
	std::cout << std::endl;

	sort(main_list, a, b);

	for (auto element : main_list){
		std::cout << element << " ";
	}
	std::cout << std::endl;


}

File: 2_semester/Laba_4/task_2.py
'''

С клавиатуры вводятся два числа a, b (1 ≤ 𝑎 ≤ 𝑏 ≤ 𝑛). Необходимо отсортировать по
неубыванию все элементы с номерами от a до b, после чего вывести на экран.

'''

from random import *


def generator(count):
	main_list = []
	for _ in range(count):
		main_list.append(randint(-100,100)) 
	return main_list

def sort_list(main_list, first, second):
	for i in range(first, second+1):
		for j in range(i, second+1):
			if main_list[i] < main_list[j]:
				main_list[i], main_list[j] = main_list[j], main_list[i]

def main():
	count = 10 #int(input(Enter n: ))
	first = 2  #int(input(Enter a: ))
	second= 8  #int(input(Enter b: ))

	main_list = generator(count)

	for element in main_list:
		print(element, end = ' ')
	print()

	sort_list(main_list, first, second)

	for element in main_list:
		print(element, end = ' ')
	print()



if __name__ == "__main__":
	main()